### PR TITLE
drop 3.7 support

### PR DESCRIPTION
**Pull request by alangenfeld**
end of lifed a couple weeks ago
https://devguide.python.org/versions/

## How I Tested These Changes

bk

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* master
  * **PR #15234** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15234" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15234?utm_source=stack-comment).

**Review by rexledesma — DISMISSED** [review]
👋

**Comment by alangenfeld** [comment]
re @johannkm 
https://github.com/dagster-io/dagster/blame/master/python_modules/automation/automation/docker/images/dagster-k8s/versions.yaml
needs updating before this / with this

**Comment by alangenfeld** [comment]
will land after https://github.com/dagster-io/dagster/pull/15243

**Comment by erinkcochran87** [comment]
@alangenfeld There are a few places in the docs that mention this - should they be added to this PR, or would you prefer I open up a new PR for that?

File: python_modules/dagster/README.md [hunk 1]
```diff
@@ -70,7 +71,7 @@ Dagster is built to be used at every stage of the data development lifecycle - l
 
 If you're new to Dagster, we recommend reading about its [core concepts](https://docs.dagster.io/concepts) or learning with the hands-on [tutorial](https://docs.dagster.io/tutorial).
 
-Dagster is available on PyPI and officially supports Python 3.7+.
+Dagster is available on PyPI and officially supports Python 3.8+.
```
Comment (alangenfeld) [inline python_modules/dagster/README.md:74]: [1]

**Review by alangenfeld — COMMENTED** [review]
I got [1] - I can try to get the others

**Comment by erinkcochran87** [comment]
This looks like the rest of it:

- /docs/content/community/contributing.mdx
- /docs/content/getting-started/hello-dagster.mdx
- /docs/content/dagster-cloud/deployment/serverless.mdx (Not sure about this one though - if we're doing this, does this also apply to support for Serverless?)
- /docs/content/deployment/guides/docker.mdx (showing in some examples as `python:3.7-slim` - not sure if this is applicable or if your changes will address this, but mentioning it JIC)

File: docs/content/getting-started/releases.mdx [hunk 2]
```diff
@@ -44,6 +44,16 @@ While technically the 0.y.z phase of Semantic Versioning is "anything goes", we
...
 
+# Python Version Support
+
+Each Dagster release strives to support the currently active versions of python.
+
+When a new version of python is released, Dagster will work to add support once Dagster's own core dependencies have been updated to support it.
+
+When a version of python reaches end of life, Dagster will drop support for it at the next convenient non-patch release.
+
+---
```
Comment (alangenfeld) [inline docs/content/getting-started/releases.mdx:55]: This needs some more word-smithing but I believe this is roughly how we treat python version support. 

At one point i had something like "previously released versions of Dagster remain available for use with older python versions"  but pulled it.

**Review by slopp — APPROVED** [review]
It would be cool to have some sort of table showing the last dagster release w/ support for a python version, eg
3.7, 1.3.14

**Pull request merged**